sormas-backend/src/test/java/de/symeda/sormas/backend/sormastosormas/validation/InfraValidationSoundnessTest.java

```diff
@@ -0,0 +1,572 @@
+		ResolvedTypeWithMembers resolvedTypeWithMembers = memberResolver.resolve(resolve, null, null);
+		ResolvedField dtoUnderTestField = resolvedTypeWithMembers.getMemberFields()[0];
+		DtoTree dtoTree = new DtoTree(dtoUnderTestField);
+		DtoTree walkedTree = buildDtoTree(dtoTree);
+		return extractInfraFieldPaths(walkedTree);
+	}
+
+	/**
+	 * Find all field paths we expect to be rejected by validation logic.
+	 * 
+	 * @param entityDto
+	 *            the entity object we inject the wrong infrastructure into
+	 * @param paths
+	 *            all paths leading to infrastructure fields
+	 * @return the set of all property paths we expect to turn up in validation
+	 * 
+	 */
+	private <T extends SormasToSormasShareableDto> Set<String> getExpectedPaths(SormasToSormasEntityDto<T> entityDto, List<ResolvedField[]> paths)
+		throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+		Set<String> expected = new HashSet<>();
+
+		for (ResolvedField[] path : paths) {
+			ArrayDeque<ResolvedField> queue = new ArrayDeque<>(Arrays.asList(path));
+			String caption = queue.stream().map(ResolvedMember::getName).collect(Collectors.joining("."));
+			expected.add(caption);
+			injectWrongInfra(queue, entityDto, caption);
+		}
+		return expected;
+	}
+
+	/**
+	 * Get all field paths which were rejected by validation logic.
+	 * 
+	 * @param entityDto
+	 *            the entity object we injected the wrong infrastructure into
+	 * @param validator
+	 *            the validator to be used
+	 * 
+	 * @return a list of rejected fields
+	 */
+	private <DTO extends SormasToSormasShareableDto, SHARED extends SormasToSormasEntityDto<DTO>, PREVIEW extends PseudonymizableDto> Set<String> getRejectedFields(
+		SHARED entityDto,
+		SormasToSormasDtoValidator<DTO, SHARED, PREVIEW> validator) {
+		ValidationErrors errors = validator.validateIncoming(entityDto);
+		return errors.getSubGroups()
+			.stream()
+			.map(ValidationErrorGroup::getMessages)
+			.flatMap(Collection::stream)
+			.map(ValidationErrorMessage::getArgs)
+			.flatMap(Arrays::stream)
+			.map(Object::toString)
+			.collect(Collectors.toSet());
+	}
+
+	private <T, DTO extends SormasToSormasShareableDto, SHARED extends SormasToSormasEntityDto<DTO>, PREVIEW extends PseudonymizableDto> void assertValidation(
+		SHARED entity,
+		DtoRootNode<T> rootNode,
+		SormasToSormasDtoValidator<DTO, SHARED, PREVIEW> validator)
+		throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+		List<ResolvedField[]> paths = getInfraPaths(rootNode);
+		Set<String> expected = getExpectedPaths(entity, paths);
+		Set<String> foundFields = getRejectedFields(entity, validator);
+		// smoke test, in case both are empty some reason this will blow up
+		assertFalse(foundFields.isEmpty());
+		assertFalse(expected.isEmpty());
+		assertEquals(new ArrayList<String>(), CollectionUtils.disjunction(foundFields, expected));
```

Comment:
```suggestion
		assertTrue(CollectionUtils.disjunction(foundFields, expected).isEmpty());
```
also remove the import

Reply:
used a message